build/teamcity-check.sh

```diff
@@ -1,7 +1,15 @@
 #!/usr/bin/env bash
 set -exuo pipefail
-build/builder.sh make check 2>&1 | go-test-teamcity
+# All packages need to be installed before we can run (some) of the checks
+# and code generators reliably. More precisely, anything that using
+# x/tools/go/loader is fragile (this includes stringer, vet and others).
```

Comment:
It does imply that, but we'd still need it here for stability (no reason we wouldn't reorder `make check` and `go generate`). I like @dt's idea though.